cadc-sia/src/main/java/ca/nrc/cadc/sia2/impl/ServiceAvailability.java

```diff
@@ -130,27 +146,60 @@ public void setState(String string)
     {
         //no-op
     }
-    
+
+    /**
+     * Obtain the base TAP URL to use.  This method will read in an optionally configured 'tapURI' property and if it
+     * appears to be a URL, then assume an unregistered TAP service was configured and treat it as the base URL,
+     * otherwise attempt to use it as a URI and look the URL up in the Registry.
+     *
+     * @return  URL     Base URL of the TAP service to use.
+     * @throws Exception    If a URL cannot be created from the specified string.
+     */
+    public static URL getTapBaseURL() throws Exception
+    {
+        RegistryClient regClient = new RegistryClient();
+        URI configuredTapURI = URI.create(ServiceAvailability.getTapURI());
+
+        if (configuredTapURI.getScheme().startsWith("http"))
+        {
+            return configuredTapURI.toURL();
+        }
+        else
+        {
+            AuthMethod am = AuthenticationUtil.getAuthMethod(AuthenticationUtil.getCurrentSubject());
+            return regClient.getServiceURL(configuredTapURI, Standards.TAP_10, (am == null) ? AuthMethod.ANON : am);
+        }
+    }
+
+
     public static String getTapURI()
     {
         if (tapURI == null)
         {
-            String fname = "SiaRunner.properties";
+            final String configFileLocation = System.getProperty("user.home") + "/config/SiaRunner.properties";
             try
             {
-                URL url = SiaRunner.class.getResource(fname);
-                if (url == null)
-                    url = SiaRunner.class.getResource("/" + fname);
-                Properties props = new Properties();
+                final URL url = new URL("file://" + configFileLocation);
+                final Properties props = new Properties();
```

Comment:
Why change the lookup from classpath to file location?

Reply:
(If we go this route, there's a PropertiesReader file that will do this work for you)

Reply:
To make it externally configurable.  This PR is meant to support ALMA, who pushes their releases through QA which uses new release hostnames each time.  It would be much easier to have it externally configurable rather than rebuilding the Docker Image each time.